combat-trainer.lic

```diff
@@ -1853,12 +1870,14 @@ class ManipulateProcess
     @manip_to_train = settings.manipulate_only_to_train
     echo("  @manip_to_train: #{@manip_to_train}") if $debug_mode_ct
 
+    Flags.add('ct-manip-broke', 'You feel your link with .* suddenly break', 'You feel your link suddenly break', 'You feel your link with .* weakening with distance.')
+
     @last_manip = Time.now - 200
   end
 
   def execute(game_state)
     return if game_state.danger || @threshold.nil? || game_state.construct_mode?
-    @filtered_npcs = game_state.npcs
+    @filtered_npcs = DRRoom.npcs
```

Comment:
Why is this variable named `filtered_npcs`? Is the `game_state` keeping track of things differently than the simple `DRRoom.npcs`? Maybe there are some `ingored_npcs` or something we're then not accounting for? Something to consider.

Reply:
Filtered npcs is the list with ignored npcs removed.

Reply:
Filtered npcs is the list with ignored npcs removed.

Reply:
game_state.npcs is drroom.npcs - ignored_npcs
ignored_npcs are npcs we dont wanr empaths making attacks on

only other way i can think to do it is make a list of construct and undead npcs and filter the room npcs through that

Reply:
Is it the case that you don't mind manipulating the ignored_npcs? I know I use those for player-created entities like familiars and watchsouls, or just NPC's that wander some hunting areas.

Reply:
Maybe better implementation would be to have a second list for npcs we want to manipulate? So manipulate npcs list will add npcs back in to filtered list

Reply:
So far in my short hunting experience, every area I've tried has had npcs I can't attack but can manipulate. Haven't had the issue where I tried to manipulate a familiar etc. I would think we just need to capture the message for that and add it to the construct check. You will try to manipulate them once a hunt then it will ignore them again

Reply:
You also can't manipulate undead.  Familiars are automatically in the ignore_npc: list in base.  If not then you will try to manipulate/app them.  I can get you any messaging you want to see.

Reply:
Don't believe undead are in the ignore_npc list, not manipulating undead is on the player. The familiars being in the base ignore list is nice then, it wasn't before, thanks for that.

Reply:
Just confused at what the benefit is of using the DRRoom.npcs over the game_state.npcs? You will encounter familiar problems eventually, once WM's join you, and a lot of HLC empaths group, so that would be a consistent problem.

Reply:
For whatever its worth, empath warriors, watchsouls, ranger spawn, and familiars are definitely annoying.  And you don't always notice right away they're messing with you.  I don't use the various pets on my necro but I imagine they would also be on that list though more of an edge case as empaths aren't often hunting with necro.

Reply:
Added a new line to base.yaml for adding back in NPCs that you want to manipulate.
going to test it now on my empath.

Reply:
Sorry, can you explain the reason behind just not using game_state.npcs? You're adding a second list now?

Reply:
Second list allows us to keep ignoring npcs we dont want to attack, and manipulate npcs we do.